src/components/modals/base/index.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from '@material-ui/core';
+
+import Button from '../../buttons';
+
+const BaseModal = ({
+  title,
+  contentText,
+  openButtonProps,
+  actionButtonsProps,
+}) => {
+  const [isModalOpen, setModalOpen] = useState(false);
+  const {
+    openButtonName,
+    openButtonOnClick,
+    ...openButtonOtherProps
+  } = openButtonProps;
+
+  const handleOpen = () => {
+    setModalOpen(true);
+  };
+
+  const handleClose = () => {
+    setModalOpen(false);
+  };
+
+  const handleOnClickOpen = onClickFunc => {
+    handleOpen();
+
+    if (onClickFunc != null) {
+      onClickFunc();
+    }
+  };
+
+  const handleOnClickClose = onClickFunc => {
+    handleClose();
+
+    if (onClickFunc != null) {
+      onClickFunc();
+    }
+  };
+
+  return (
+    <div>
+      <Button
+        onClick={() => handleOnClickOpen(openButtonOnClick)}
+        {...openButtonOtherProps}
+      >
+        {openButtonName}
+      </Button>
+
+      <Dialog open={isModalOpen} onClose={handleClose}>
+        <DialogTitle>{title}</DialogTitle>
+
+        <DialogContent>
+          <DialogContentText>{contentText}</DialogContentText>
+        </DialogContent>
+
+        <DialogActions>
+          {actionButtonsProps.map(buttonProps => {
```

Comment:
I think it would be more idiomatic to render the actionbuttons as children - that way don't have to pass in the props but just directly render the children.

Reply:
PM me on slack if this isn't clear

Reply:
No this is fine. I think this would be better as well. It simplifies the code for BaseModal : )

Reply:
Didn't think of this before, thanks for mentioning it! It makes BaseModal a lot more flexible

Reply:
Hmm I'm running into a problem with this though. If I just use props.children for DialogActions, then I can't easily make it so that whenever the user clicks on an action button, it closes the modal. You know any solution or workaround for this?

Reply:
ah I did not think of that - you can use a render function a la https://frontarm.com/james-k-nelson/passing-data-props-children/

Reply:
So I can expect props.children to be a function then? If that's the case then this would work: `<DialogActions>
          {props.children(onClickFunc => handleOnClickClose(onClickFunc))}
        </DialogActions>`

Reply:
yep - BaseDialog expects the props.children stuff which is instantiated in components like ButtonWithConfirm etc